sk0-back/src/users/controllers/users.controller.spec.ts

```diff
@@ -2,41 +2,99 @@ import { Test, TestingModule } from '@nestjs/testing';
 import { UsersController } from './users.controller';
 import { AuthService } from '../services/auth.service';
 import { LoginUserDto } from '../dtos/login-user.dto';
-import { NotFoundException, UnauthorizedException } from '@nestjs/common';
+import {
+  NotFoundException,
+  UnauthorizedException,
+  ConflictException,
+} from '@nestjs/common';
+import { UsersService } from '../services/users.service';
+import { CreateUserDto } from '../dtos/create-user.dto';
+import { User } from '../user.entity';
 
 describe('UsersController', () => {
-  let usersController: UsersController;
-  let authService: AuthService;
+  let controller: UsersController;
+  let mockAuthService: Partial<AuthService>;
+  let mockUsersService: Partial<UsersService>;
 
   beforeEach(async () => {
+    mockAuthService = {
+      createUser: jest.fn(),
+      loginUser: jest.fn(),
+    };
+
     const module: TestingModule = await Test.createTestingModule({
       controllers: [UsersController],
       providers: [
-        {
-          provide: AuthService,
-          useValue: {
-            loginUser: jest.fn(),
-          },
-        },
+        { provide: UsersService, useValue: mockUsersService },
+        { provide: AuthService, useValue: mockAuthService },
       ],
     }).compile();
 
-    usersController = module.get<UsersController>(UsersController);
-    authService = module.get<AuthService>(AuthService);
+    controller = module.get<UsersController>(UsersController);
+    mockAuthService = module.get<AuthService>(AuthService);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+
+  // Register
+  describe('createUser', () => {
+    const createUserDto: CreateUserDto = {
+      name: 'Test User',
+      email: 'a@bc.de',
+      password: 'password',
+    };
+
+    const createdUserMock: User = {
+      id: 1,
+      name: 'Test User',
+      email: 'a@bc.de',
+      password: 'password',
+    };
+
+    it('should create a new user successfully', async () => {
+      jest
+        .spyOn(mockAuthService, 'createUser')
+        .mockResolvedValue(createdUserMock);
+
+      const result = await controller.createUser(createUserDto);
+
+      expect(result.email).toEqual(createdUserMock.email);
+      expect(mockAuthService.createUser).toHaveBeenCalledTimes(1);
+      expect(mockAuthService.createUser).toHaveBeenCalledWith(createUserDto);
+    });
+
+    it('should throw ConflictException', async () => {
+      jest
+        .spyOn(mockAuthService, 'createUser')
```

Comment:
Then you try to spy on it. Choose one of the mocking styles.
jest.fn().mockResolved() or jest.spyOn() - but in this case if know it well, you don't have to mock the services outside the scope.